.github/workflows/semantic-pr-check.yaml

```diff
@@ -0,0 +1,38 @@
+---
+name: "Semantic PR Title Check"
+# Ensure that the PR title conforms to the Conventional Commits and our choice of types and scopes, so that library version bumps can be detected automatically
+
+on:
+  pull_request_target:
+    types: [opened, edited, synchronize]
+
+permissions:
+  pull-requests: read
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            chore
+            ci
+            docs
+            feat
+            fix
+            perf
+            refactor
+          scopes: |
+            harness
```

Comment:
We're just going to allow `harness` for now though, right? Per discussion of https://github.com/canonical/operator/pull/1252/files#diff-9769a3f5504af3cd735e08f078b94f827bdd51eacac2e9d80ac9761bc485171fR181

Reply:
Let me merge this in my fork to see what the meaning of `scopes: [a,b,c]; requireScope: false` is, that is if arbitrary, unlisted scopes are allowed.

Reply:
done, added a sane catch-all